src/cyclotom.c

```diff
@@ -698,14 +698,16 @@ Obj             Cyclotomic (
                 /* is just the inverse transformation of 'ConvertToBase'   */
                 for ( i = 0; i < n; i += p ) {
                     cof = res[(i+n/p)%n];
-                    res[i] = INTOBJ_INT( - INT_INTOBJ(cof) );
                     if ( ! IS_INTOBJ(cof)
                       || (cof == INTOBJ_INT(-(1L<<NR_SMALL_INT_BITS))) ) {
                         CHANGED_BAG( STATE(ResultCyc) );
                         cof = DIFF( INTOBJ_INT(0), cof );
```

Comment:
Still think it would make sense to change `cof = DIFF( INTOBJ_INT(0), cof );` to `cof = AINV( cof );`, matching the code in line ca. 982, but alas, that's certainly not a requirement for this PR, which already contains many other nice improvements :-).